[[ Bug 22322 ]] Fix graphical artefacts after removing focus from a group on mobile

livecodeian opened the pull request:
This patch fixes bug 22322, where removing focus from a group could cause graphical artefacts when using acceleratedRendering in combination with the "Motif" lookandfeel

runrevmark commented:
Wouldn't a simpler fix be just to override gettransient() in MCGroup and return 0... Since groups never render a focus border, so their transient rect never changes?

livecodeian committed:
[[ Bug 22322 ]] Add release note for bug 22322

This patch adds a release note for bug 22322, where removing focus from a group
could cause graphical artefacts when using acceleratedRendering

livecodeian committed:
[[ TileCache ]] Fix tiles composited at wrong offset when group focus changed

This patch fixes an issue where changes to group keyboard focus which change the
effective rect of the group would not trigger an update of the layer rect,
leading to a mismatch between the previously cached tiles and the offset at
which they are rendered.

As groups don't render a focus border, the effective rect should not change when
focused.

livecodeian commented:
@runrevmark updated

livecodeali commented:
@livecode-vulcan review ok 76b901a

livecode-vulcan commented:
💙 review by @livecodeali ok 76b901a523f96c39ee44f16efaba6a675877bf9c

livecode-vulcan commented:
😎 test success 76b901a523f96c39ee44f16efaba6a675877bf9c

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/630)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/39)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/39)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/39)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/39)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2156)
* try-community-universal-ios-iphoneos12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/381)
* try-community-universal-ios-iphonesimulator12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/377)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2185)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2172)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2171)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2205)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2145)
